testing/ef-tests/src/cases/blockchain_test.rs

```diff
@@ -310,9 +310,11 @@ fn run_case(
         // Compute and check the post state root
         let hashed_state =
             HashedPostState::from_bundle_state::<KeccakKeyHasher>(output.state.state());
-        let (computed_state_root, _) =
-            StateRoot::overlay_root_with_updates(provider.tx_ref(), hashed_state.clone())
-                .map_err(|err| Error::block_failed(block_number, program_inputs.clone(), err))?;
+        let (computed_state_root, _) = StateRoot::overlay_root_with_updates(
+            provider.tx_ref(),
+            &hashed_state.clone().into_sorted(),
```

Comment:
I wonder if it's more efficient to add `clone_into_sorted` that uses `iter().cloned()` then we don't clone the map metadata

https://github.com/paradigmxyz/reth/blob/82a186104311c660e77982883542754542f1f206/crates/trie/common/src/hashed_state.rs#L365-L377

Reply:
Ran benchmarks comparing `clone().into_sorted()` vs `clone_into_sorted`:

**HashedPostState** (accounts x slots):
| Case | Improvement |
|------|-------------|
| 64x0 | ~2.2% faster |
| 64x8 | ~2.8% faster |
| 64x64 | ~1.3% faster |
| 256x8 | ~2-4% faster |
| 256x64 | ~1% faster |
| 1024x64 | ~1-2% faster |

**HashedStorage** (slots):
| Case | Improvement |
|------|-------------|
| dirty 16 | ~20% slower |
| wiped 16 | ~31% faster |
| dirty 128 | ~2% faster |
| wiped 1024 | ~3% faster |
| dirty 1024 | flat (~-1%) |

**Summary**: `clone_into_sorted` wins in most realistic sizes. Only small dirty-storage (16 slots) shows a regression, likely noise at that scale.

[Benchmark SVG reports (gist)](https://gist.github.com/yongkangc/d3c9b5294154c94f1945b54f037dbcc7)